runelite-client/src/main/java/net/runelite/client/plugins/statusbars/BarRenderer.java

```diff
@@ -0,0 +1,180 @@
+import java.util.function.Supplier;
+
+@RequiredArgsConstructor
+class BarRenderer
+{
+	private static final Color BACKGROUND = new Color(0, 0, 0, 150);
+	private static final Color OVERHEAL_COLOR = new Color(216, 255, 139, 150);
+	private static final int OVERHEAL_OFFSET = 2;
+	private static final int HEAL_OFFSET = 3;
+	private static final int ICON_AND_COUNTER_OFFSET_X = 1;
+	private static final int ICON_AND_COUNTER_OFFSET_Y = 21;
+	private static final int SKILL_ICON_HEIGHT = 35;
+	private static final int COUNTER_ICON_HEIGHT = 18;
+	private static final int OFFSET = 2;
+	private static final int WIDTH = 20;
+	private static final int PADDING = 1;
+	private final StatusBarsConfig config;
+	private final Supplier<Integer> maxValueSupplier;
+	private final Supplier<Integer> currentValueSupplier;
+	private final Supplier<Integer> healSupplier;
+	private final Supplier<Color> colorSupplier;
+	private final Supplier<Color> healColorSupplier;
+	private final Supplier<Image> iconSupplier;
+	private int maxValue;
+	private int currentValue;
+
+	private void refreshSkills()
+	{
+		maxValue = maxValueSupplier.get();
+		currentValue = currentValueSupplier.get();
+	}
+
+	/**
+	 * Renders a status bar along with its restoration bar(s), icons and counters as appropriate
+	 * @param graphics Graphics.
+	 * @param x The location on the client where it will draw the bar on the x axis starting on the left side.
+	 * @param y The location on the client where it will draw the bar on the y axis starting on the bottom side.
+	 * @param height The height of the bar.
+	 * @param leftSide Is used to indicate the rendering the left side of the bar, used for the counters.
+	 */
+	void renderBar(Graphics2D graphics, int x, int y, int height, boolean leftSide)
+	{
+		final int filledHeight = getBarHeight(maxValue, currentValue, height);
+		final Color fill = colorSupplier.get();
+
+		refreshSkills();
+
+		graphics.setColor(BACKGROUND);
+		graphics.drawRect(x, y, WIDTH - PADDING, height - PADDING);
+		graphics.fillRect(x, y, WIDTH, height);
+
+		graphics.setColor(fill);
+		graphics.fillRect(x + PADDING,
+			y + PADDING + (height - filledHeight),
+			WIDTH - PADDING * OFFSET,
+			filledHeight - PADDING * OFFSET);
+
+		if (config.enableRestorationBars())
+		{
+			renderRestore(graphics, x, y, height);
+		}
+
+		if (config.enableSkillIcon() || config.enableCounter())
+		{
+			renderIconsAndCounters(graphics, x, y, leftSide);
+		}
+	}
+
+	private void renderIconsAndCounters(Graphics2D graphics, int x, int y, boolean leftSide)
+	{
+		String counterText = Integer.toString(currentValue);
+		final int widthOfCounter = graphics.getFontMetrics().stringWidth(counterText);
+		final int centerText = (WIDTH - PADDING) / 2 - (widthOfCounter / 2);
+		final Image icon = iconSupplier.get();
+		int offset = 1;
+
+		graphics.setFont(FontManager.getRunescapeSmallFont());
+		graphics.setColor(Color.WHITE);
+
+		if (config.enableCounter())
+		{
+			if (leftSide)
+			{
+				offset = 2;
```

Comment:
Why is this offset one pixel greater when `leftSide` is true? Shouldn't the offset be the same regardless of which side the bar was drawn on?

Reply:
This is fixed now, it now correctly draws the counters in the center for both bars.